roboto/bot.py

```diff
@@ -825,6 +829,97 @@ async def send_contact(
             await make_request(self.session, HTTPMethod.POST, '/sendContact', request),
         )
 
+    async def send_poll(
+        self,
+        chat_id: Union[ChatID, str],
+        question: str,
+        options: List[str],
+        is_anonymous: Optional[bool] = None,
+        poll_type: Optional[PollType] = None,
```

Comment:
Options and poll_type have incompatible types with the `Poll` class

Reply:
Yes, they do. I cannot create a Poll with a `voter_count` already set in an option, that would not make sense. Input options are just strings. This matches the API reference.

About `PollType`: We are striving for compatibility. Code using an old version of Roboto should __not__ break if the Telegram API is changed. Therefore:

- If I ensure the user is __sending__ a `PollType` and a new poll type is introduced in the API, that bot's code will remain valid. The user will not be able to send the newer poll types unless he updates Roboto and/or changes his code, but that's expected. His bot won't break, all is good and stronger types ensured better code ever since the beginning.
- If I try to __validate__ server responses and a new poll type is introduced, as soon as the bot receives one of those, it breaks. That's bad.

Of course, the first case would break as well if a poll type is __removed__, but that is unlikely and would probably go through a deprecation cycle, since it would break the API itself.

Reply:
Oh, I see, that's very cool indeed.